BirkhoffErgodicThm.lean

```diff
@@ -366,3 +366,21 @@ theorem birkhoffErgodicTheorem  (hf : MeasurePreserving f μ μ) (hφ : Integrab
   rw [inv_lt_iff_one_lt_mul₀ (Nat.cast_pos.mpr k.succ_pos)]
   norm_num at hk' ⊢
   linarith
+
+/-- Here we drop the assumption that the observable is `Measurable`. -/
+theorem birkhoffErgodicTheorem' {Φ : α → ℝ} (hf : MeasurePreserving f μ μ) (hΦ : Integrable Φ μ) :
+    ∀ᵐ x ∂μ, Tendsto (birkhoffAverage ℝ f Φ · x) atTop (𝓝 (invCondexp μ f Φ x)) := by
+  -- Take `φ` as the measurable approximation to the ae measurable `Φ`.
+  let φ := hΦ.left.mk
+  have hφ' : Measurable φ := hΦ.left.measurable_mk
+  have hΦ' : Φ =ᵐ[μ] φ := hΦ.left.ae_eq_mk
+  have hφ : Integrable φ μ := (integrable_congr hΦ.left.ae_eq_mk).mp hΦ
+  -- Obtain a full measure set such that the three relevant results hold.
+  obtain ⟨s, hs, hs'⟩ : ∃ s ∈ ae μ, Set.EqOn (invCondexp μ f Φ) (invCondexp μ f φ) s :=
+    eventuallyEq_iff_exists_mem.mp <| condExp_congr_ae hΦ'
+  obtain ⟨t, ht, ht'⟩ := eventually_iff_exists_mem.mp <| birkhoffErgodicTheorem μ hf hφ hφ'
+  have := ae_all_iff.mpr <| birkhoffAverage_ae_eq_of_ae_eq ℝ hf.quasiMeasurePreserving hΦ'
+  obtain ⟨u, hu, hu'⟩ := eventually_iff_exists_mem.mp this
+  -- Apply the three results on the chosen set.
+  refine eventually_iff_exists_mem.mpr ⟨s ∩ t ∩ u, inter_mem (inter_mem hs ht) hu, fun y hy ↦ ?_⟩
+  simp [hs' hy.1.1, ht' y hy.1.2, hu' y hy.2]
```

Comment:
Perhaps we should rename the previous statement `_aux` or prime and have this one be called `birkhoffErgodicTheorem`?